compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -1860,6 +1872,30 @@ static void buildDefaultReadWriteFunctions(AggregateType* ct) {
     normalize(fn);
   }
 
+  //
+  // Keep generating the 'encodeTo' method so that the override versions don't
+  // cause errors.
+  //
+  if (makeEncodeTo && !hasEncodeTo) {
+    ArgSymbol* fileArg = NULL;
+    FnSymbol* fn = buildWriteThisFnSymbol(ct, &fileArg, "encodeTo");
+
+    // Compiler generated versions of readThis/writeThis now throw.
+    fn->throwsErrorInit();
+
+    if (fUseIOFormatters) {
+      if (hasWriteThis) {
+        fn->insertAtTail(new CallExpr("writeThis", gMethodToken, fn->_this, fileArg));
```

Comment:
Interesting. Is this intended to be transitional, to help people migrate? In that case I would expect we would print a warning here?

Reply:
This is intended to be transitional in order to help people migrate, but I feel like we need a real story there. For now this was simplest. I didn't put a warning because that would make it more difficult to track how many tests were failing. Should we add a warning that is controlled by some flag, e.g. ``--warn-translated-writeThis``?

Related to this - I imagine most users would probably just change ``writeThis`` to ``encodeTo``, and leave it at that. While the formatter API will hopefully be flexible across different formatters, ultimately it requires type authors to opt-in when they implement an ``encodeTo``. We'll need to develop some good examples to demonstrate the preferred usage to users.

Reply:
Yes, I understand the testing point. Could you add a comment to indicate that we probably want to have a warning here, and make sure to bring it up in follow-up issue about  the relationship between Encoder/Decoder and readThis/writeThis?